HIP/hip-657.md

```diff
@@ -53,45 +53,38 @@ The use of the METADATA key to update NFT metadata will not affect the functiona
 ## Specification
 
 
-The METADATA key will be a new key that allows for updating the metadata of an NFT in a new NftMetadataUpdate transaction.
+The METADATA key will be a new key that allows for updating the metadata of an NFT in a new UpdateNft transaction.
 
-The NftMetadataUpdate transaction should accept the following parameters:
-	- identifier (string): the 0.0.<token-type>.<serial-no>
-- the new NFT metadata: the new metadata for the NFT
+[HIP-646](https://hips.hedera.com/hip/hip-646) and [HIP-765](https://hips.hedera.com/hip/hip-765) will add metadata on the token level. The METADATA key will also allow the metadata to be updated on the token level.
 
+The UpdateNft transaction should accept the following parameters:
 
-The NftMetadataUpdate transaction should have a new token type field called "dynamic_metadata" of type bool, which is valid only for non-fungible tokens. By default, this field is set to false (meaning the metadata for this token type is immutable). If set to true, owners of this token type can change their NFT's metadata via the NftMetadataUpdate transaction.
+- `nft_id`: the identifier (`NftID`) of the NFT to be updated
+- `metadata`: the new metadata for the NFT
 
 
 Example transaction:
 ```js
-NftMetadataUpdate()
-     .setTokenId(tokenId)
-     .setMetaData(newMetaData)
+const transaction = await TokenUpdateNft()
+     .setNftId(nftId)
+     .setMetadata(newMetadata)
      .sign(METADATAKEY)
      .execute(client);
 ```
 
 
-Protobuff
+Protobuf
 ```
-message NftMetadataUpdateTransactionBody {
+message TokenUpdateNftTransactionBody {
     /**
-     * The Token to be updated
+     * The identifier of the NFT to be updated
      */
-    TokenID token = 1;
-
-    /**
-     * The metadata key of the Token. If Token is immutable, transaction will resolve to
-     * TOKEN_IS_IMMUTABlE.
-     */
-     
-     Key metadataKey = 2;
+    NftID nft_id = 1;
```

Comment:
Believe this is the first time that `NftId` is being used by HAPI, so probably we should also define it in this HIP. I think it is only used by services state currently.

Reply:
Would it be easier for implementation / less of an anti-pattern if the format matched burning tokens where the tokenId and the serial numbers are set in the transaction instead?
https://docs.hedera.com/hedera/sdks-and-apis/sdks/token-service/burn-a-token

Reply:
This path would also allow multiple NFTs that need to be updated to the same metadata to be executed in one transaction submition, better developer experience.

Reply:
I think this is an excellent idea @ty-swirldslabs. Will discuss it with the team.

Reply:
I have updated the parameter list to match `TokenBurn`.